src/main/java/com/hp/octane/plugins/bamboo/octane/BambooPluginServices.java

```diff
@@ -175,21 +191,33 @@ public CIServerInfo getServerInfo() {
     @Override
     public void stopPipelineRun(String pipeline, CIParameters ciParameters) {
         log.info("starting pipeline stop");
+
         Callable<String> action = () -> {
             BambooUserManager um = ComponentLocator.getComponent(BambooUserManager.class);
             BambooUser user = um.getBambooUser(getRunAsUser());
-            ImmutableChain chain = planMan.getPlanByKey(PlanKeys.getPlanKey(pipeline.toUpperCase()), ImmutableChain.class);
+            PlanKey planKey = PlanKeys.getPlanKey(pipeline.toUpperCase());
+            ImmutableChain chain = planMan.getPlanByKey(planKey, ImmutableChain.class);
 
             if (chain != null) {
                 if (!isUserHasPermission(BambooPermission.BUILD, user, chain)) {
                     throw new PermissionException(HttpStatus.SC_FORBIDDEN);
                 }
 
                 log.info(String.format("plan key=%s ,build key=%s ,chain key=%s", chain.getPlanKey().getKey(), chain.getBuildKey(), chain.getKey()));
-                try {
+
+                CIParameter octaneExecutionId = ciParameters.getParameters().stream()
+                        .filter(parameter -> parameter.getName().equals(OCTANE_AUTO_ACTION_EXECUTION_ID))
+                        .findFirst().orElse(null);
+
+                if (octaneExecutionId == null) {
                     planExecMan.stopPlan(chain.getPlanKey(), true, user.getName());
-                } catch (InterruptedException e) {
```

Comment:
I saw that method throw this exception
why have removed this exception catching?

Reply:
I will add exception catching

Reply:
@nissimshitrit Actually, there is no ability to add this exception
![image](https://user-images.githubusercontent.com/71069933/167633706-8322b1a3-aaba-4ea9-b30b-0941a1eb5c08.png)